checker.c

```diff
@@ -1,21 +1,117 @@
 #include <stdio.h>
 #include <assert.h>
+#include<stdbool.h>
+#include "checker.h"
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
+void DisplaybatteryStatus(float attribute_value, int IdxMessage)
+{
+    if (Sel_Language == ENGLISH)
+    {
+        printf("%f %s\n",attribute_value, BatteryStatus_English[IdxMessage]);
+    }
+    else{
+        printf("%f %s\n",attribute_value,BatteryStatus_German[IdxMessage]);
+    }
+}
+
+int LowerThresholdChecker(float attribute_value, float lowerthreshold)
+{
+  bool isValid = attribute_value < lowerthreshold;
+  if(isValid)
+  {
+    DisplaybatteryStatus(attribute_value,0); 
+    return 0; 
+  }
+    
+  else return 1;
+}
+
+int Check_LowAttributeWarning(float attribute_value,float lowerthreshold_warning, float lowerthreshold)
+{
+   if((attribute_value<=lowerthreshold_warning) && (attribute_value > lowerthreshold))
+   {
+       DisplaybatteryStatus(attribute_value,2);
```

Comment:
What can you do to clarify the index 2 at the place of calling? How can a developer avoid mistakes in the index while calling?